pyipn/universe.py

```diff
@@ -172,3 +173,47 @@ def from_yaml(cls, yaml_file):
                 universe.register_detector(det)
 
             return universe
+
+    def calculate_annulus(self, detector1, detector2):
+        d1, d2 = self._detectors[detector1], self._detectors[detector2]
+        dxyz = (d2.location.get_cartesian_coord().xyz -
+                d1.location.get_cartesian_coord().xyz)
+        
+        #calculate ra and dec of vector d  pointing from detector1 to detector2
+        dcart = Location(SkyCoord(x=dxyz[0], y=dxyz[1], z=dxyz[2], representation_type='cartesian', unit='km'))
+        norm_d = dcart.get_norm_vec(u.km)
+        ra = dcart.coord.represent_as(UnitSphericalRepresentation).lon
+        dec = dcart.coord.represent_as(UnitSphericalRepresentation).lat
+
+        #calculate angle theta between center point d and annulus
+        distance = np.linalg.norm(dxyz) * u.km
+        dt = (self._T0[list(self._detectors.keys()).index(detector1)] -
+              self._T0[list(self._detectors.keys()).index(detector2)]) * u.s
+        #rounding to 15th decimal because small numerical errors cause issues with numbers slightly over 1
+        theta = np.arccos(np.around((constants.c * dt / distance).decompose().value, 15))
```

Comment:
another way to do this is to just truncate the number. Keep a note that we need to discuss this.